### ISSUE TITLE
False positive in Angular for lists inside a for loop

### ISSUE THREAD

**Issue by CSharpFiasco** [dequelabs/axe-core#4871]
### Product

axe Linter

### Product Version

4.10.9

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation
Linter should allow `<li>` elements in a `<ul>` element inside a for loop in Angular

#### Actual
Linter flags with the following message:
`axe-linter (list): <ul> and <ol> must only directly contain <li>, <script> or <template> elements`

#### How to Reproduce
```html
<ul>
  @for (bullet of bullets(); track bullet) {
    <li class="education-text-bullet">{{bullet}}</li>
  }
</ul>
```

#### Additional context
The [ngForOf](https://angular.dev/api/common/NgForOf) structural directive has been deprecated and so the [@for](https://angular.dev/api/core/@for) block is the recommended syntax.

This also occurs for @if block.

```html
<ul>
  @if (true) {
    <li class="education-text-bullet">test</li>
  }
</ul>
```
